lib/marisa/grimoire/vector/bit-vector.cc

```diff
@@ -782,7 +782,7 @@ void BitVector::build_index(const BitVector &bv,
       const std::size_t unit_num_0s =
           std::min<std::size_t>(bits_remaining, MARISA_WORD_SIZE) - unit_num_1s;
 
-      const std::size_t zero_bit_id = (~num_0s + 1) % 512;
+      const std::size_t zero_bit_id = (0 - num_0s) % 512;
```

Comment:
As 0 is signed, num_0s will by converted to signed and 0 - num_0s will be signed. This is certain to be a signed underflow, which should be avoided as it is undefined behavior.

Hence, this change is not an improvement.

Reply:
`signed op unsigned -> unsigned`

https://en.cppreference.com/w/cpp/language/operator_arithmetic#Conversions